sonar-dependency-check-plugin/src/main/js/report_page/components/DependencyCheckReportApp.js

```diff
@@ -21,13 +21,21 @@ import React from "react";
 // SonarComponents (referenced as sonar-components here, see the Webpack config)
 // exposes React components exposed by SonarQube.
 import { DeferredSpinner } from "sonar-components";
+import { isBranch, isPullRequest } from "sonar-helpers";
 import { getJSON } from "sonar-request";
 
 export function findDependencyCheckReport(options) {
-  return getJSON("/api/measures/component", {
-      component : options.component.key,
-      metricKeys : "report"
-    }).then(function(response) {
+  var request = {
+    component : options.component.key,
+    metricKeys : "report"
+  };
+  if (isBranch(options.branchLike)) {
+    request.branch = options.branchLike.name;
+  } else if (isPullRequest(options.branchLike)) {
```

Comment:
Why not set both request keys?

Reply:
Because SonarQube complains with "Either branch or pull request can be provided, not both"